src/Tool/Infra/AuthenticatingHttpClient.cs

```diff
@@ -24,6 +24,11 @@ public Task<Stream> GetStreamAsync(string url, CancellationToken cancellationTok
         return RetryLoopOnUnauthorized(url, 3, token => HttpGetStreamWithUsefulException(url, token), cancellationToken);
     }
 
+    public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken = default)
+    {
+        return RetryLoopOnUnauthorized(request.RequestUri, 3, token => http.SendAsync(request, token), cancellationToken);
```

Comment:
is there some sort of backoff retry or just an instant back to back to back?

Reply:
It does Console prompts for username/password (expand the diff downward in the file to see) and then recreates the HttpClient with the new auth settings. So there's definitely a backoff retry it's just controlled by how fast the user can type. :wink: